stat/stat.go

```diff
@@ -0,0 +1,59 @@
+package stat
+
+import (
+	"sync"
+	"time"
+)
+
+type Stat struct {
+	// requestTimes is a map of hour -> method -> count
+	requestTimes map[int]map[string]int
+	// initTimes is a map of hour -> time
+	initTimes  map[int]time.Time
+	requestsMu *sync.Mutex
+}
+
+// day is the number of nanoseconds in a day
+const day = 24 * time.Hour
+
+func NewStat() Stat {
```

Comment:
I think it is more idiomatic go to call this `New` (since it will be called as `stat.New` and `Stat` is the only type defined in the package).